zcash_client_sqlite/src/wallet.rs

```diff
@@ -1086,6 +1086,33 @@ pub(crate) fn block_fully_scanned(
     }
 }
 
+pub(crate) fn block_max_scanned(
+    conn: &rusqlite::Connection,
+) -> Result<Option<BlockMetadata>, SqliteClientError> {
+    conn.query_row(
+        "SELECT blocks.height, hash, sapling_commitment_tree_size, sapling_tree
+         FROM blocks
+         JOIN (SELECT MAX(height) AS height FROM blocks) blocks_max
+         ON blocks.height = blocks_max.height",
```

Comment:
Wouldn't it be more performant to use `ORDER BY height DESC LIMIT 1`?

Reply:
I honestly have no idea about the relative performance profiles of these two queries. It depends upon whether `ORDER BY` is lazy, and how `MAX` interacts with the index on `height`.